Source/Details/_ASDisplayView.mm

```diff
@@ -358,14 +359,79 @@ - (void)tintColorDidChange
   [node tintColorDidChange];
 }
 
-- (BOOL)canBecomeFirstResponder {
-  ASDisplayNode *node = _asyncdisplaykit_node; // Create strong reference to weak ivar.
-  return [node canBecomeFirstResponder];
+#pragma mark UIResponder Handling
+
+#define HANDLE_RESPONDER_METHOD(__sel) \
+  ASDisplayNode *node = _asyncdisplaykit_node; \
+  SEL sel = @selector(__sel); \
+  /* Prevent an infinite loop in here if [super canBecomeFirstResponder] was called on a
+  / _ASDisplayView subclass */ \
+  if (ASSubclassOverridesSelector([_ASDisplayView class], [self class], sel)) { \
+    /* Check if we can call through to ASDisplayNode subclass directly */ \
+    if (ASDisplayNodeSubclassOverridesSelector([node class], sel)) { \
+      return [node __sel]; \
+    } else { \
+    /* Call through to views superclass as we expect super was called from the
+      _ASDisplayView subclass and a node subclass does not overwrite canBecomeFirstResponder */ \
+      return [self __##__sel]; \
+    } \
+  } else { \
+    /* Call through to internal node __canBecomeFirstResponder that will consider the view in responding */ \
+    return [node __##__sel]; \
+  } \
+
+
+- (BOOL)canBecomeFirstResponder
+{
+  HANDLE_RESPONDER_METHOD(canBecomeFirstResponder);
```

Comment:
I think you could use _cmd for each of these. They could also be made a macro for IMPLEMENT_RESPONDER_METHOD instead of HANDLE_, and define the method as well as its contents.

Reply:
Moved all into one macros and called it `IMPLEMENT_RESPONDER_METHOD`